File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(idx,result):
    global count
    if idx == n-1:
        if result - arr[idx] == t:
            count+=1
        elif result + arr[idx] == t:
            count+=1
        return 0
    dfs(idx+1,result+arr[idx])
    dfs(idx+1,result-arr[idx])
    return 0

def solution(numbers, target):
    global t
    global arr
    global count
    global n
    n = len(numbers)
    count = 0
    t = target
    arr = numbers
    dfs(0,0)
    return count

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int result = 0;

void dfs(int depth, int tmp, vector<int>numbers, int target){
    // 종료 조건
    if(depth == numbers.size()){
        if(tmp == target){
            result ++;
            return;
        }
        return;
    }
    dfs(depth+1, tmp+numbers[depth],numbers, target);
    dfs(depth+1, tmp-numbers[depth], numbers, target);
    
}
int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    dfs(0, 0, numbers, target);
    answer = result;
    return answer;
}